aten/src/ATen/templates/TensorBody.h

```diff
@@ -394,13 +447,174 @@ class CAFFE2_API Tensor {
+  /// and never the `grad_accumulator`. For the latter, use
+  /// `set_grad_accumulator`. This allows late construction of an interior
+  /// `Variable`.
+  void set_gradient_edge(torch::autograd::Edge edge) const noexcept;
+
+  // Versions
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  /// Increments the version count of this `Variable`.
+  void bump_version() const noexcept {
+    impl_->bump_version();
+  }
+  void set_version_counter(const c10::VariableVersion& version_counter) const noexcept {
+    impl_->set_version_counter(version_counter);
+  }
+
+  /// Retrieves this `Variable`s version counter.
+  const c10::VariableVersion& version_counter() const noexcept {
+    return impl_->version_counter();
+  }
+
+  /// Retrieves the current value of the `Variable`'s version counter.
+  /// Equivalent to calling `version_counter().current_version()`.
+  uint32_t current_version() const noexcept {
+    return impl_->version_counter().current_version();
+  }
+
+  // Autograd Graph Interaction
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  /// Update the `grad_fn` of an existing Variable. Called after in-place
+  /// modifications.
+  ///
+  /// For View Variables:
+  /// Called after in-place modifications. Modifies the grad_fn of the base
+  /// Variable.
+  void rebase_history(torch::autograd::Edge gradient_edge) const;
+
+  // Hooks
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  void add_hook(std::shared_ptr<torch::autograd::FunctionPreHook> hook) const;
+  const std::vector<std::shared_ptr<torch::autograd::FunctionPreHook>>& hooks() const noexcept;
+  void clear_hooks() const;
+
+  template <typename T>
+  using hook_return_void_t = c10::guts::enable_if_t<std::is_void<typename std::result_of<T&(Tensor)>::type>::value, unsigned>;
+  template <typename T>
+  using hook_return_var_t = c10::guts::enable_if_t<std::is_same<typename std::result_of<T&(Tensor)>::type, Tensor>::value, unsigned>;
+  // Remove hook at given position
+  void remove_hook(unsigned pos) const;
+
+  // Returns the index of the hook in the list which can be used to remove hook
+  // Register a hook with no return value
+  template <typename T>
+  hook_return_void_t<T> register_hook(T&& hook) const;
+  // Register a hook with variable return value
+  template <typename T>
+  hook_return_var_t<T> register_hook(T&& hook) const;
+
+  // View Variables
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  /// Returns true if this `Variable` is a view of another `Variable`.
+  bool is_view() const noexcept;
+
+  /// Returns the `Variable` that this `Variable` is a view of. If this
+  /// `Variable` is not a view, throw a `std::runtime_error`.
+  const Tensor& base() const;
+
+  // Miscellaneous
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  void set_name(const std::string& name) const;
```

Comment:
Is that actually used?

Reply:
Yeah, I think ONNX uses it to associate names with variables on export